mitodl/mit-open
# add Articles api

# What are the relevant tickets?
API for #109 

# Description (What does it do?)
This PR implements a very simple `StaffPost` model and API for creation/editing of rich-text content (e.g., staff writes a highlight of newly released courses). 

# How can this be tested?
1. As a staff user, visit http://localhost:8063/api/v1/staff_posts/
2. POST some data, e.g., `{ "html": "<p>Hello world</p>", "title": "First Post" }`. and then try viewing it at http://localhost:8063/api/v1/staff_posts/POST_ID
3. Try deleting/updating posts, and sending suspicious HTML like `<script>` tags. It should be sanitized.
    - The sanitization is overly aggressive right now, but I'd like to tweak that in a subsequent PR and allow more discussion on https://github.com/mitodl/hq/discussions/2600
